app/main.py

```diff
@@ -1,2 +1,80 @@
+from typing import Generator
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self._size = 8
+        self._threshold = 0.3
+        self._table: list[list] = [[] for _ in range(self._size)]
+        self._resize_const = 2
+
+    def print_table(self) -> None:
+        print(self._table)
+
+    def __getitem__(self, key: any) -> all:
+        hash_key = hash(key)
+        table_index = hash_key % self._size
+
+        for _ in range(self._size):
+            entry = self._table[table_index]
+            if entry and entry[1] == key:
+                return entry[0]
+            table_index = next(self._gen_next_slot(table_index))
```

Comment:
The collision handling logic in the `__getitem__` method should ensure that it continues searching for the key until it either finds it or confirms that the key is not present. Consider using a loop that iterates through all possible slots.